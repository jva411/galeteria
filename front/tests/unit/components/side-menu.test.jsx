/* eslint-disable no-undef */
import React from 'react'
import renderer from 'react-test-renderer'
import SideMenu from '../../../components/side-menu'
import { render, screen } from '@testing-library/react'

import * as nextRouter from 'next/router';

/* eslint-disable no-import-assign */
nextRouter.useRouter = jest.fn();
nextRouter.useRouter.mockImplementation(() => ({ pathname: '/' }));

describe('unit test of component \'side-menu\'', () => {

    test('should render logo images to control via css', () => {
        render(<SideMenu/>)

        expect(screen.getByTitle('logo-lenovo1')).toBeInTheDocument()
        expect(screen.getByTitle('logo-lenovo2')).toBeInTheDocument()
    })

    test('should render dashboard link with proper href location already being clicked', () => {
        render(<SideMenu/>)

        expect(screen.getByText(/dashboard/iu).getAttribute('href')).toBe(null)
    })

    test('should render models link with proper href location', () => {
        render(<SideMenu/>)

        expect(screen.getByText(/models/iu).getAttribute('href')).toBe('/models')
    })

    test('should render datasets link with proper href location', () => {
        render(<SideMenu/>)

        expect(screen.getByText(/datasets/iu).getAttribute('href')).toBe('/datasets')
    })

    test('should render users link with proper href location', () => {
        render(<SideMenu/>)

        expect(screen.getByText(/users/iu).getAttribute('href')).toBe('/users')
    })

    test('should render groups link with proper href location', () => {
        render(<SideMenu/>)

        expect(screen.getByText(/groups/iu).getAttribute('href')).toBe('/groups')
    })

    test('should SideMenu match with Snapshot', () => {
        const tree = renderer.create(<SideMenu/>).toJSON()

        expect(tree).toMatchSnapshot()
    })
})