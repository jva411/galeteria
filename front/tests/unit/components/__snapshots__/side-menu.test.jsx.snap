// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit test of component 'side-menu' should SideMenu match with Snapshot 1`] = `
<div
  className="css-1443u2l"
  data-testid="sidemenu-pane"
>
  <img
    src="/logo-lenovo.png"
    title="logo-lenovo1"
  />
  <img
    src="/logo-lenovo-mw.png"
    title="logo-lenovo2"
  />
  <div
    className="css-j7qwjs"
  >
    <div
      className="chakra-linkbox css-10n1vic"
    >
      <svg
        className="chakra-icon css-1jmtuae"
        fill="currentColor"
        focusable={false}
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3 13h8V3H3v10zm0 8h8v-6H3v6zm10 0h8V11h-8v10zm0-18v6h8V3h-8z"
        />
      </svg>
      <a
        className="chakra-linkbox__overlay css-1hnz6hu"
        href={null}
        name="Dashboard"
      >
        Dashboard
      </a>
    </div>
    <div
      className="chakra-linkbox css-10n1vic"
    >
      <svg
        className="chakra-icon css-1jmtuae"
        fill="currentColor"
        focusable={false}
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4 6H2v14c0 1.1.9 2 2 2h14v-2H4V6zm16-4H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-1 9H9V9h10v2zm-4 4H9v-2h6v2zm4-8H9V5h10v2z"
        />
      </svg>
      <a
        className="chakra-linkbox__overlay css-1hnz6hu"
        href="/models"
        name="Models"
      >
        Models
      </a>
    </div>
    <div
      className="chakra-linkbox css-10n1vic"
    >
      <svg
        className="chakra-icon css-1jmtuae"
        fill="currentColor"
        focusable={false}
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M3 12h4v9H3v-9zm14-4h4v13h-4V8zm-7-6h4v19h-4V2z"
          />
        </g>
      </svg>
      <a
        className="chakra-linkbox__overlay css-1hnz6hu"
        href="/datasets"
        name="Datasets"
      >
        Datasets
      </a>
    </div>
    <div
      className="chakra-linkbox css-10n1vic"
    >
      <svg
        className="chakra-icon css-1jmtuae"
        fill="currentColor"
        focusable={false}
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 11c1.66 0 2.99-1.34 2.99-3S17.66 5 16 5c-1.66 0-3 1.34-3 3s1.34 3 3 3zm-8 0c1.66 0 2.99-1.34 2.99-3S9.66 5 8 5C6.34 5 5 6.34 5 8s1.34 3 3 3zm0 2c-2.33 0-7 1.17-7 3.5V19h14v-2.5c0-2.33-4.67-3.5-7-3.5zm8 0c-.29 0-.62.02-.97.05 1.16.84 1.97 1.97 1.97 3.45V19h6v-2.5c0-2.33-4.67-3.5-7-3.5z"
        />
      </svg>
      <a
        className="chakra-linkbox__overlay css-1hnz6hu"
        href="/users"
        name="Users"
      >
        Users
      </a>
    </div>
    <div
      className="chakra-linkbox css-10n1vic"
    >
      <svg
        baseProfile="tiny"
        className="chakra-icon css-1jmtuae"
        fill="currentColor"
        focusable={false}
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        version="1.2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 14c1.381 0 2.631-.56 3.536-1.465.904-.904 1.464-2.154 1.464-3.535s-.56-2.631-1.464-3.535c-.905-.905-2.155-1.465-3.536-1.465s-2.631.56-3.536 1.465c-.904.904-1.464 2.154-1.464 3.535s.56 2.631 1.464 3.535c.905.905 2.155 1.465 3.536 1.465zM20 15c.69 0 1.315-.279 1.768-.731.453-.452.732-1.077.732-1.769 0-.69-.279-1.315-.732-1.768-.453-.453-1.078-.732-1.768-.732-.691 0-1.316.279-1.769.732-.452.453-.731 1.078-.731 1.768 0 .691.279 1.316.731 1.769s1.078.731 1.769.731zM20 15.59c-1.331 0-2.332.406-2.917.968-1.115-.917-2.878-1.558-5.083-1.558-2.266 0-3.995.648-5.092 1.564-.596-.565-1.608-.974-2.908-.974-2.188 0-3.5 1.09-3.5 2.182 0 .545 1.312 1.092 3.5 1.092.604 0 1.146-.051 1.623-.133l-.04.27c0 1 2.406 2 6.417 2 3.762 0 6.417-1 6.417-2l-.02-.255c.463.073.995.118 1.603.118 2.051 0 3.5-.547 3.5-1.092 0-1.092-1.373-2.182-3.5-2.182zM4 15c.69 0 1.315-.279 1.768-.732.453-.453.732-1.078.732-1.768 0-.689-.279-1.314-.732-1.768-.453-.452-1.078-.732-1.768-.732-.691 0-1.316.28-1.769.732-.452.454-.731 1.079-.731 1.768 0 .69.279 1.315.731 1.768.453.453 1.078.732 1.769.732z"
        />
      </svg>
      <a
        className="chakra-linkbox__overlay css-1hnz6hu"
        href="/groups"
        name="Groups"
      >
        Groups
      </a>
    </div>
  </div>
</div>
`;
